src/errors.rs

```diff
@@ -12,4 +12,12 @@ pub enum SessionError {
     HTTP(#[from] http::Error),
     #[error("unknown Session store error")]
     Unknown,
+    #[error("Generic Database insert error")]
+    GenericInsertError,
```

Comment:
these should be `GenericInsertError(String)`  and the error map can include a {1} which would display the inner string on display.